middleware/middleware.go

```diff
@@ -72,6 +73,23 @@ func NewEchoAuthMiddleware(secret []byte, bypassAuth bool) echo.MiddlewareFunc {
 	})
 }
 
+func environmentInCache(authRepo domain.AuthRepo, claims *domain.Claims) bool {
+	envID := claims.Environment
+	key := claims.APIKey
+	keys, err := authRepo.GetKeysForEnvironment(context.Background(), envID)
+	if err != nil {
+		return false
+	}
+
+	// check if apikey is in cache
+	for _, k := range keys {
+		if key == k {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Maybe I'm being dumb here, but can we not just do this? 

```go
// Just check if the API key exists in the cache, if it doesn't then we must have removed it
key, err := authRepo.Get(ctx, claims.Key)
if err != nil {
   if errors.Is(err, domain.ErrCacheNotFound) {
        return false
   }
   return false
 }
 
 return true
```

Reply:
```
func isKeyInCache(repo keyLookUp, claims *domain.Claims) bool {
	key := claims.APIKey
	_, ok := repo.Get(context.Background(), domain.AuthAPIKey(key))
	if ok {
		return true
	}
	return false
}
```